### PR TITLE
Cherrypick commits from  the "20231222_rke2_100_clusters_1vs3_nodes_comparison" branch

### PR DESCRIPTION
**Pull request by fgiudici**
Pick relevant commits to be ported on master.
Fulll list of evaluated commits  below ("OK" --> cherry-picked on master, "S" --> commit skipped).

```
OK	d403edb editorial changes
OK	3838709 Results: share main recap page and all raw data
OK	cbd64c4 Results: share spreadsheet with all the test results
OK	d1e9876 Load test data
S	96390d5 Azure: use the Standard_E8ads_v5 vm size for Rancher nodes
OK	9050928 AKS: allow to specify different vm sizes for different node pools
OK	1f331b7 Use shorter timeouts
S	e421497 Revert parallelism change because SSH firealling
S	66ece51 quality of life: use more threads against APIs
OK	66b05d5 draft: split running tests from getting access commands
OK	a472a0f azure: add support for serial console
S	4cc0621 rke2 test: change prefix, use more downstream clusters, change location
OK	054c70b azure: use broader 172.20.0.0/16 network for downstream clusters
S	c22c737 k6 tests: add more resources
S	5963c8d Increase AKS downstream clusters to 50
S	9012e75 run_test_auto.sh: update script for batch testing
OK	0a17bfe k6:Roles&Users: make the resource load more gentle
OK	5e68e41 azure_host: use Premium Disk by default
OK	b1c5eef azure_host: mount ephemeral disk on /data only if big enough
OK	3252d52 azure_host: fix ephemeral OS option
OK	f96733e tests: fix creation of configmaps
OK	3dcbc5f bugfix: use correct port
OK	Fix run_tests.mjs
S	8b12c9c Set upstream cluster nodes to Standard_D4ds_v5
S	48259d7 load testing: create more downstream clusters
S	c4c97a0 load testing: create more users
OK	75696b8 load test: add more and bigger configmaps and secrets upstream
S	f5b3125 k6 test: script to automate creation and retrieval of results
S	3b11a37 terraform inputs: change tunnel ports for RKE2
OK	26da662 README: add note how to destroy all tunnels
S!	79e05ca docs: add basic outline
S	91d5882 Change zones to avoid bumping into limits
OK	b547a16 aks: use better tier by default
OK	02a3476 azure: bump provider to avoid 404 errors
OK	ec9ecf4 azure/aks: support ephemeral disks for OS
S	4375b4a use different prefix for aks
S	4aa41e9 create restricted admins exclusively
S	8d224ec drop loading monitoring in downstream
S	4e7cd1c add instructions to run benchmarks manually
OK	315626b run_tests.mjs: output Kubernetes API as well
OK	b59f7a5 run_tests.mjs: make output clearer
S	3dc6f13 bump load on upstream cluster
S	be333d9 drop loading downstream clusters with secrets and configmaps
S	a26e6b1 drop cgroups exporters

```

### CODE REVIEW

File: bin/run_tests.mjs [hunk 1]
```diff
@@ -13,11 +13,11 @@ import {k6_run} from "./lib/k6.mjs"
...
-const CONFIG_MAP_COUNT = 1000
-const SECRET_COUNT = 1000
-const ROLE_COUNT = 10
-const USER_COUNT = 5
-const PROJECT_COUNT = 20
+const ROLE_COUNT = 20
+const CONFIG_MAP_COUNT = 30_000
+const SECRET_COUNT = 30_000
+const USER_COUNT = 50
+const PROJECT_COUNT = 50
```
Comment (moio) [inline bin/run_tests.mjs:20]: I'd suggest resource count changes to stay in the test branch.

File: k6/create_k8s_resources.js [hunk 2]
```diff
@@ -7,7 +7,7 @@ import * as k8s from './k8s.js'
 const namespace = "scalability-test"
 const configMapCount = Number(__ENV.CONFIG_MAP_COUNT)
 const secretCount = Number(__ENV.SECRET_COUNT)
-const data = encoding.b64encode("a".repeat(10*1024))
+const data = encoding.b64encode("a".repeat(1000*1024))
```
Comment (moio) [inline k6/create_k8s_resources.js:10]: I'd suggest resource size changes to stay in the test branch.

**Review by moio — APPROVED** [review]
With most commits being already peer reviewed during development, I only left a couple of comments.

This PR is a bit of a grab bag, but I don't feel like opening separate PRs with all commits being already separated out brings a lot of benefit.

Thanks for taking care of this

File: bin/run_tests.mjs (hunk 1 above)
Comment (fgiudici) [reply bin/run_tests.mjs]: Yep, it slept in, good catch. I  left anyway the code loading ConfigMaps and Secrets on the upstream cluster too (which is the reason that made me including the commit).

**Pull request merged**